app/containers/Ringer/index.tsx

```diff
@@ -1,41 +1,34 @@
 import { Audio } from 'expo-av';
 import React, { useEffect, useRef } from 'react';
-import { View } from 'react-native';
 
 export enum ERingerSounds {
 	DIALTONE = 'dialtone',
 	RINGTONE = 'ringtone'
 }
 
 const Ringer = React.memo(({ ringer }: { ringer: ERingerSounds }) => {
-	const sound = useRef<Audio.Sound | null>(null);
+	const sound = useRef(new Audio.Sound());
+
 	useEffect(() => {
-		(async () => {
-			let expo = null;
-			switch (ringer) {
-				case ERingerSounds.DIALTONE:
-					expo = await Audio.Sound.createAsync(require(`./dialtone.mp3`));
-					break;
-				case ERingerSounds.RINGTONE:
-					expo = await Audio.Sound.createAsync(require(`./ringtone.mp3`));
-					break;
-				default:
-					expo = await Audio.Sound.createAsync(require(`./dialtone.mp3`));
-					break;
+		const loadAndPlay = async () => {
+			try {
+				const soundFile = ringer === ERingerSounds.DIALTONE ? require(`./dialtone.mp3`) : require(`./ringtone.mp3`);
+				await sound.current.loadAsync(soundFile);
+				await sound.current.playAsync();
+				await sound.current.setIsLoopingAsync(true);
+			} catch (error) {
+				console.error('Error loading sound:', error);
 			}
-			sound.current = expo.sound;
-			await sound.current.playAsync();
-			await sound.current.setIsLoopingAsync(true);
-		})();
-	}, []);
+		};
 
-	useEffect(() => () => stopSound(), []);
+		loadAndPlay();
 
-	const stopSound = () => {
-		sound?.current?.unloadAsync();
-	};
+		return () => {
+			sound.current?.unloadAsync();
+		};
+	}, []);
 
-	return <View />;
+	return null;
```

Comment:
Considering that the Ringer don't need to be a view to reproduce the sound, can change this behavior? wdyt?

Reply:
No, because when using a hook its instance will always exist, being a component its instance ceases to exist when the component is disassembled.

Reply:
Ok